### PR TITLE
planner: update documents related to Plan Cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@qw4990 Is this PR related to a v6.0 change? If yes, is there a FD JIRA for it?
@ran-huang FYI

File: system-variables.md [hunk 1]
```diff
@@ -987,6 +987,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 这个变量用来设置并行 hash aggregation 算法 partial 阶段的执行并发度。对于聚合函数参数不为 distinct 的情况，HashAgg 分为 partial 和 final 阶段分别并行执行。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_ignore_close_stmt_cmd` <span class="version-mark">从 v6.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 这个变量用来设置是否忽略关闭 Prepared Statement 的指令。
+- 如果开关打开，Binary 协议的 `COM_STMT_CLOSE` 信号和文本协议的 `DEALLOCATE PREPARE` 都会被忽略。
+
```
Comment (TomShawn) [inline system-variables.md:996]:
```suggestion
```

File: system-variables.md [hunk 2]
```diff
@@ -987,6 +987,13 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 这个变量用来设置并行 hash aggregation 算法 partial 阶段的执行并发度。对于聚合函数参数不为 distinct 的情况，HashAgg 分为 partial 和 final 阶段分别并行执行。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_ignore_close_stmt_cmd` <span class="version-mark">从 v6.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:990]:
```suggestion
### `tidb_ignore_close_stmt_cmd`（从 v6.0 版本开始引入）
```

File: sql-prepare-plan-cache.md [hunk 3]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
 +------------------------+
 1 row in set (0.00 sec)
 ```
+
+# 手动清空计划缓存
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:135]:
```suggestion
## 手动清空计划缓存
```

File: sql-prepare-plan-cache.md [hunk 4]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+TiDB 支持手动清空计划缓存，方式为 `admin flush session/instance plan_cache`，`session` 和 `instance` 用于指定需要清空的缓存级别；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:137]:
```suggestion
你可手动清空计划缓存，方式为使用 `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` 语句， 其中 `SESSION` 和 `INSTANCE` 用于指定需要清空的缓存级别。
```

File: sql-prepare-plan-cache.md [hunk 5]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+MySQL [test]> select @@last_plan_from_cache; -- 由于缓存被情况此时无法再次命中
++------------------------+
+| @@last_plan_from_cache |
++------------------------+
+|                      0 |
++------------------------+
+1 row in set (0.00 sec)
+```
+
+目前 TiDB 暂不支持清空 `global` 级别的计划缓存，即一次性清空整个集群的计划缓存，使用时会报错：
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:179]:
```suggestion
目前 TiDB 暂不支持清空 `global` 级别的计划缓存，即不支持一次性清空整个集群的计划缓存，使用时会报错：
```

File: sql-prepare-plan-cache.md [hunk 6]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+{{< copyable "sql" >}}
+
+```sql
+MySQL [test]> admin flush global plan_cache;
+ERROR 1105 (HY000): Do not support the 'admin flush global scope.'
+```
+
+# 忽略 `COM_STMT_CLOSE` 指令和 `DEALLOCATE PREPARE` 语句
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:188]:
```suggestion
## 忽略 `COM_STMT_CLOSE` 指令和 `DEALLOCATE PREPARE` 语句
```

File: sql-prepare-plan-cache.md [hunk 7]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
 +------------------------+
 1 row in set (0.00 sec)
 ```
+
+## 手动清空计划缓存
+
+你可手动清空计划缓存，方式为使用 `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` 语句， 其中 `SESSION` 和 `INSTANCE` 用于指定需要清空的缓存级别。
```
Comment (Reminiscent) [inline sql-prepare-plan-cache.md:137]:
```suggestion
你可手动清空计划缓存，方式为使用 `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` 语句，在不指定作用域的时候，默认作用域为 SESSION，其中 `SESSION` 和 `INSTANCE` 用于指定需要清空的缓存级别。
```

**Review by Reminiscent — APPROVED** [review]

File: sql-prepare-plan-cache.md [hunk 8]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
 +------------------------+
 1 row in set (0.00 sec)
 ```
+
+## 手动清空计划缓存
+
+你可手动清空计划缓存，方式为使用 `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` 语句，在不指定作用域的时候，默认作用域为 SESSION，其中 `SESSION` 和 `INSTANCE` 用于指定需要清空的缓存级别。
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:137]:
```suggestion
通过执行 `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` 语句，你可以手动清空计划缓存。语句种的作用域用于指定需要清空的缓存级别，可以为 `SESSION` 或 `INSTANCE`，默认为 `SESSION`。
```

File: sql-prepare-plan-cache.md [hunk 9]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+MySQL [test]> prepare stmt from 'select * from t';
+Query OK, 0 rows affected (0.00 sec)
+
+MySQL [test]> execute stmt;
+Empty set (0.00 sec)
+
+MySQL [test]> execute stmt;
+Empty set (0.00 sec)
+
+MySQL [test]> select @@last_plan_from_cache; -- 可以命中缓存
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:156]:
```suggestion
MySQL [test]> select @@last_plan_from_cache; -- 选择计划缓存
```

File: sql-prepare-plan-cache.md [hunk 10]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
++------------------------+
+1 row in set (0.00 sec)
+
+MySQL [test]> admin flush session plan_cache; -- 清空当前 session 的计划缓存
+Query OK, 0 rows affected (0.00 sec)
+
+MySQL [test]> execute stmt;
+Empty set (0.00 sec)
+
+MySQL [test]> select @@last_plan_from_cache; -- 由于缓存被情况此时无法再次命中
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:170]:
```suggestion
MySQL [test]> select @@last_plan_from_cache; -- 由于缓存被清空，此时无法再次选中
```

File: sql-prepare-plan-cache.md [hunk 11]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
++------------------------+
+| @@last_plan_from_cache |
++------------------------+
+|                      0 |
++------------------------+
+1 row in set (0.00 sec)
+```
+
+目前 TiDB 暂不支持清空 `global` 级别的计划缓存，即不支持一次性清空整个集群的计划缓存，使用时会报错：
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:179]:
```suggestion
TiDB 暂不支持清空 `GLOBAL` 级别的计划缓存，即不支持一次性清空整个集群的计划缓存，使用时会报错：
```

File: system-variables.md [hunk 12]
```diff
@@ -987,6 +987,13 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 这个变量用来设置并行 hash aggregation 算法 partial 阶段的执行并发度。对于聚合函数参数不为 distinct 的情况，HashAgg 分为 partial 和 final 阶段分别并行执行。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_ignore_prepared_cache_close_stmt`（从 v6.0 版本开始引入）
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 这个变量用来设置是否忽略关闭 Prepared Statement 的指令。
+- 如果开关打开，Binary 协议的 `COM_STMT_CLOSE` 信号和文本协议的 `DEALLOCATE PREPARE` 都会被忽略。
```
Comment (shichun-0415) [inline system-variables.md:995]:
```suggestion
- 如果变量值设为 `ON`，Binary 协议的 `COM_STMT_CLOSE` 信号和文本协议的 `DEALLOCATE PREPARE` 语句都会被忽略。
```

File: sql-prepare-plan-cache.md [hunk 13]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+mysql> set @@tidb_ignore_prepared_cache_close_stmt=1;  -- 打开开关
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> prepare stmt from 'select * from t'; -- 第一次 prepare
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> execute stmt;                        -- 第一次 execute
+Empty set (0.00 sec)
+
+mysql> deallocate prepare stmt;             -- 第一次执行完后立即释放
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:231]:
```suggestion
mysql> deallocate prepare stmt;             -- 第一次 execute 后立即释放
```

File: sql-prepare-plan-cache.md [hunk 14]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+{{< copyable "sql" >}}
+
+```sql
+MySQL [test]> admin flush global plan_cache;
+ERROR 1105 (HY000): Do not support the 'admin flush global scope.'
+```
+
+## 忽略 `COM_STMT_CLOSE` 指令和 `DEALLOCATE PREPARE` 语句
+
+Prepared Statement 推荐的使用方式是，Prepare 一次，然后 Execute 多次，最后 Close，如：
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:190]:
```suggestion
Prepared Statement 推荐的使用方式是，prepare 一次，然后 execute 多次，最后 close，如：
```

File: sql-prepare-plan-cache.md [hunk 15]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+```sql
+MySQL [test]> prepare stmt from '...'; -- 第一次 prepare
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 一次使用后立即释放
+MySQL [test]> prepare stmt from '...'; -- 第二次 prepare
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 再次释放
+```
+
+这样的使用方式会让第一次执行得到的计划被立即清理，不能在第二次被复用；
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:215]:
```suggestion
这样的使用方式会让第一次执行得到的计划被立即清理，不能在第二次被复用。
```

File: sql-prepare-plan-cache.md [hunk 16]
```diff
@@ -131,3 +131,117 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+为了兼容这样的使用方式，TiDB 支持 `tidb_ignore_prepared_cache_close_stmt` 变量，打开后会忽略掉关闭 Preapre Statement 的信号，解决上述问题，如：
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:217]:
```suggestion
为了兼容这样的使用方式，TiDB 支持 `tidb_ignore_prepared_cache_close_stmt` 变量，打开后会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 17]
```diff
@@ -987,6 +987,13 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 这个变量用来设置并行 hash aggregation 算法 partial 阶段的执行并发度。对于聚合函数参数不为 distinct 的情况，HashAgg 分为 partial 和 final 阶段分别并行执行。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_ignore_prepared_cache_close_stmt`（从 v6.0 版本开始引入）
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 这个变量用来设置是否忽略关闭 Prepared Statement 的指令。
+- 如果变量值设为 `ON`，Binary 协议的 `COM_STMT_CLOSE` 信号和文本协议的 `DEALLOCATE PREPARE` 语句都会被忽略。
```
Comment (qiancai) [inline system-variables.md:995]: 这里可以加一个链接指向 “忽略 `COM_STMT_CLOSE` 指令和 `DEALLOCATE PREPARE` 语句” 这个章节吗？

Comment (qw4990) [reply system-variables.md]: COM_STMT_CLOSE 我们的文档内还没有相关内容；DEALLOCATE 的连接已经补充；

File: sql-prepare-plan-cache.md [hunk 18]
```diff
@@ -131,3 +131,119 @@ MySQL [test]> select @@last_plan_from_cache;
...
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 一次使用后立即释放
+MySQL [test]> prepare stmt from '...'; -- 第二次 prepare
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 再次释放
+```
+
+这样的使用方式会让第一次执行得到的计划被立即清理，不能在第二次被复用。
+
+为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 `tidb_ignore_prepared_cache_close_stmt` 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```
Comment (qiancai) [inline sql-prepare-plan-cache.md:219]:
```suggestion
为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 [`tidb_ignore_prepared_cache_close_stmt`](#tidb_ignore_prepared_cache_close_stmt从-v60-版本开始引入) 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```

File: sql-prepare-plan-cache.md [hunk 19]
```diff
@@ -131,3 +131,119 @@ MySQL [test]> select @@last_plan_from_cache;
...
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 一次使用后立即释放
+MySQL [test]> prepare stmt from '...'; -- 第二次 prepare
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 再次释放
+```
+
+这样的使用方式会让第一次执行得到的计划被立即清理，不能在第二次被复用。
+
+为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 [`tidb_ignore_prepared_cache_close_stmt`](#tidb_ignore_prepared_cache_close_stmt从-v60-版本开始引入) 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:219]:
```suggestion
为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 [`tidb_ignore_prepared_cache_close_stmt`](system-variables.md#tidb_ignore_prepared_cache_close_stmt从-v60-版本开始引入) 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```

File: sql-prepare-plan-cache.md [hunk 20]
```diff
@@ -131,3 +131,119 @@ MySQL [test]> select @@last_plan_from_cache;
...
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 一次使用后立即释放
+MySQL [test]> prepare stmt from '...'; -- 第二次 prepare
+MySQL [test]> execute stmt using ...;
+MySQL [test]> deallocate prepare stmt; -- 再次释放
+```
+
+这样的使用方式会让第一次执行得到的计划被立即清理，不能在第二次被复用。
+
+为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 [`tidb_ignore_prepared_cache_close_stmt`](system-variables.md#tidb_ignore_prepared_cache_close_stmt从-v60-版本开始引入) 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:219]:
```suggestion
为了兼容这样的使用方式，从 v6.0 起，TiDB 支持 [`tidb_ignore_prepared_cache_close_stmt`](/system-variables.md#tidb_ignore_prepared_cache_close_stmt从-v60-版本开始引入) 变量。打开该变量后，TiDB 会忽略关闭 Preapre Statement 的信号，解决上述问题，如：
```

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**